modules/light-clients/07-tendermint/types/update.go

```diff
@@ -162,12 +162,62 @@ func checkTrustedHeader(header *Header, consState *ConsensusState) error {
 	return nil
 }
 
-// checkValidity checks if the Tendermint header is valid.
-// CONTRACT: consState.Height == header.TrustedHeight
-func checkValidity(
-	clientState *ClientState, consState *ConsensusState,
-	header *Header, currentTimestamp time.Time,
+// VerifyClientMessage checks if the clientMessage is of type Header or Misbehaviour and validates the message
+func (cs *ClientState) VerifyClientMessage(
+	ctx sdk.Context, clientStore sdk.KVStore, cdc codec.BinaryCodec, consState *ConsensusState,
+	header exported.ClientMessage,
 ) error {
+	switch header.(type) {
+	case *Header:
+		return verifyHeader(ctx, cs, clientStore, cdc, consState, header, ctx.BlockTime())
```

Comment:
Maybe `cs.verifyHeader`? ie make client state function

Reply:
Yeah, I originally had it this way but I wasn't sure if we wanted to expose this or not.

Reply:
I like it being an unexposed client state function